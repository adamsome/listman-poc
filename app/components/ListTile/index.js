import React from 'react';

import './ListTile.scss'

const ListTile = (props) => {
  return (
    <a className="list-tile box">
      <span className="list-tile__title-wrapper">
        <span className="list-tile__title title is-4">
          {props.title}
        </span>
      </span>
    </a>
  )
}

export default ListTile
