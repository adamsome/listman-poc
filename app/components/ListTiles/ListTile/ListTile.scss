.list-tile {
  padding-bottom: 0.5rem;
}
.list-tile__title-wrapper {
  display: block;
  min-height: 60px;
}

.list-tile__title {
  display: inline-block;
  overflow: hidden;
  max-height: 48px;
}

