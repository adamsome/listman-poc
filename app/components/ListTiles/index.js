import React from 'react'

import ListTile from './ListTile'

const renderListTiles = (lists) => {
  if (!lists.length) {
    return <div>No Items!</div>
  }
  return lists.map((list, index) => {
    return (
      <div key={index} className="column is-one-quarter">
        <ListTile
          key={index}
          title={list.title}
        />
      </div>
    )
  })
}

const ListTiles = ({ lists }) => (
  <div className="container">
    <h2 className="title is-2">
      Lists
    </h2>
    <div className="columns is-multiline">
      {renderListTiles(lists)}
    </div>
  </div>
)

export default ListTiles

