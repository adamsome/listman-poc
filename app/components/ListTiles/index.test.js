import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import ListTiles from './index'
import ListTile from './ListTile'

it('renders null based on initial state (empty "list" array)', () => {
  const wrapper = shallow(<ListTiles lists={[]}/>)
  expect(wrapper.contains(<div>No Items!</div>)).toBe(true)
  expect(toJson(wrapper)).toMatchSnapshot()
})

it('renders <div> w/ correct children of type <ListTile>', () => {
  const title = 'Test One'
  const wrapper = shallow(<ListTiles lists={[ { title } ]}/>)
  expect(wrapper.find(ListTile)).toHaveLength(1)

  const child = wrapper.find(ListTile).first()
  const childProps = child.props()
  expect(childProps.title).toEqual(title)
  expect(toJson(wrapper)).toMatchSnapshot()
})

