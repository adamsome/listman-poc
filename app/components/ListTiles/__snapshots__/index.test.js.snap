exports[`test renders <div> w/ correct children of type <Item> 1`] = `
<div
  className="container">
  <h2
    className="title is-2">
    Lists
  </h2>
  <div
    className="columns is-multiline">
    <div
      className="column is-one-quarter">
      <ListTile
        title="Test One" />
    </div>
  </div>
</div>
`;

exports[`test renders null based on initial state (empty "list" array) 1`] = `
<div
  className="container">
  <h2
    className="title is-2">
    Lists
  </h2>
  <div
    className="columns is-multiline">
    <div>
      No Items!
    </div>
  </div>
</div>
`;
